### DISCUSSION TITLE
Asserting that object is "empty"

### DISCUSSION THREAD

**Discussion by ITaluone** [fluentassertions/fluentassertions#1862] [Ideas]
Hi

I have a DTO with quite an amount of properties. If I want to test, if this object is empty (meaning all properties at default) I get a very long list of this lines of code

```csharp

dto.FirstProperty.Should().BeNullOrEmpty();
dto.SecondProperty.Should().BeNull();
dto.ThirdProperty.Should().BeNullOrEmpty();
dto.FourthProperty.Should().BeNullOrEmpty();

```

This is quite annoying and repetitive. 

If there was a solution like

```csharp

dto.Should().BeEmpty();

// OR

dto.Should().HaveAllDefaults();

```
testing if every property has its default value, would be great.

**Comment by dennisdoomen** [comment]
No such methods exist, but why not do this:

```cs
dto.Should().BeEquivalentTo(new DTO());
```

**Reply by ITaluone** [reply]
Oh... 🙈

This solution is too easy.. I want a unnecessary complicated one.. :joy:
